js/deep.js

```diff
@@ -1,164 +1,186 @@
+                                </div>
+                                <div class="content">
+                                    <h2>${elem.title}</h2>
+                                    <p>${elem.subTitle}</p>
+                                </div>
+                            </div>
+                        ${elem.isActive === true ? '<span class="arrow-right"></span>' : ''}
+                    </a>
+                </li>`
+            ).join('')}
+        </ul>`
+    // end create deep-menu
+
+    //start create sub-menu
+    const creaSubMenu = (arg) =>
+        `<ul>
+            ${arg.map(elem =>
+            elem.children ?
+                `<li>
+                    <a href="${elem.link}">
+                        <div class="item child">
+                            <div class="icon">
+                                <i class="${elem.icon}"></i>
+                            </div>
+                            <div class="content">
+                                <h2>${elem.title}</h2>
+                            </div>
+                            <div class="show">${creaSubMenu(elem.children)}</div>
                         </div>
-                        <div class="content">
-                            <h2>${elem.title}</h2>
-                            <p>${elem.subTitle}</p>
+                        <span class="arrow arrow-right"></span>
+                    </a>
+                </li>`
+                : `<li>
+                    <a href="${elem.link}">
+                        <div class="item">
+                            <div class="icon">
+                                <i class="${elem.icon}"></i>
+                            </div>
+                            <div class="content">
+                                <h2>${elem.title}</h2>
+                            </div>
                         </div>
-                    </div>
-                <span class="${elem.isActive === true ? 'arrow-right' : ''}"></span>
-            </li>`
-    ).join('')}
-    </ul>`
+                    </a>
+                </li>`
+            ).join('')}
+        </ul>`
+    // end create sub-menu
+
+    // start create button search
+    const search =
+        `<li>
+            <div class="search">
+                <input type="text" placeholder="search..."/>
+                <i class="fas fa-search"></i>
+            </div>
+        </li>`
+    // end create button search
+
+    // start create element
+    function createElement() {
+        const container = document.createElement('div');
+        container.className = 'container';
+        const frag = document.createRange().createContextualFragment(creaDeepMenu(deepMenu));
+        container.appendChild(frag);
+        const body = document.getElementsByTagName('body')[0];
+        body.appendChild(container);
+        const searchBtn = document.createRange().createContextualFragment(search);
+        document.getElementsByTagName('ul')[0].appendChild(searchBtn);
```

Comment:
Create correct method name
Add break line for better view
```suggestion
        const container = document.createElement('div');
        container.className = 'container';
        
        const frag = document.createRange().createContextualFragment(createDeepMenu(deepMenu));
        container.appendChild(frag);
        
        const body = document.getElementsByTagName('body')[0];
        body.appendChild(container);
        
        const searchBtn = document.createRange().createContextualFragment(search);
        document.getElementsByTagName('ul')[0].appendChild(searchBtn);
```